vaadin-renderer-flow-parent/vaadin-renderer-flow/src/main/java/com/vaadin/flow/data/renderer/LitRenderer.java

```diff
@@ -201,8 +201,9 @@ private Registration createJsRendererFunction(Element container,
                     SerializableBiConsumer<SOURCE, JsonArray> handler = clientCallables
                             .get(handlerName);
                     SOURCE item = keyMapper.get(itemKey);
-
-                    handler.accept(item, args);
+                    if (item != null) {
```

Comment:
Is there a possibility that a new item will coincidentally reuse the key of one of the previous items?

Reply:
By default the `KeyMapper` uses Java object identity (hashCode) for equality comparisons. So, if you override equals/hashCode then yes: the keyMapper.get() resolves to the "new" item even when "old" item was clicked in the grid.

Also, for Grid and TreeGrid you can specify your own Key value provider, like so:
```
treeGrid.setUniqueKeyDataGenerator("key", Item::key);
```
Meaning that you can use whatever you want to specify what is the key of an object - and the keyMapper will use that.

From my point of view, two item instances with the same key (same identity) represent one entity - so it should not (theoretically) matter if you run the function handler over the "old" one or the "new" one. It might become a problem when the identity function is not defined correctly and two "not identical" objects share the same key. 

In any case, if you think this is an issue worth pursuing, I suggest creating a new bug for it - because I think this concept is not related to this bug that much...

Reply:
I agree with your reasoning 👍 Thanks for the thorough comment.